core/src/main/scala/org/apache/spark/security/SocketAuthServer.scala

```diff
@@ -66,42 +87,45 @@ private[spark] abstract class SocketAuthServer[T](
 
 }
 
+/**
+ * Create a socket server class and run user function on the socket in a background thread.
+ * This is the same as calling SocketAuthServer.setupOneConnectionServer except it creates
+ * a server object that can then be synced from Python.
+ */
+private[spark] class SocketFuncServer(
+    authHelper: SocketAuthHelper,
+    threadName: String,
+    func: Socket => Unit) extends SocketAuthServer[Unit](authHelper, threadName) {
+
+  override def handleConnection(sock: Socket): Unit = {
+    func(sock)
+  }
+}
+
 private[spark] object SocketAuthServer {
 
   /**
-   * Create a socket server and run user function on the socket in a background thread.
-   *
-   * The socket server can only accept one connection, or close if no connection
-   * in 15 seconds.
```

Comment:
please save this comment -- I guess move it to the class `SocketAuthServer`.  In particular ,its helpful to note that this only accepts one connection, its not a long-lived thing which is reused.

Reply:
Yeah that is a useful comment, I didn't intend to take this out. I'll put it back in.